airflow/providers/google/cloud/triggers/bigquery.py

```diff
@@ -89,6 +90,19 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
             },
         )
 
+    def safe_to_cancel(self) -> bool:
+        """
+        Whether it is safe to cancel the external job which is being executed by this trigger.
+
+        This is to avoid the case that `asyncio.CancelledError` is called because the trigger itself is stopped.
+        Because in those cases, we should NOT cancel the external job.
+        """
+        self.log.info("Checking if it is safe to cancel the job.")
```

Comment:
This line does not provide much information 🤔  We probably should move it to `run` and log whether it's safe and why instead.

Reply:
+1

Reply:
Removed it as its not required.